src/backend/api/Fusion.Resources.Api/Controllers/Contracts/ContractsController.cs

```diff
@@ -55,23 +52,41 @@ public async Task<ActionResult<ApiContract>> GetProjectContract([FromRoute]Proje
         }
 
         [HttpGet("/projects/{projectIdentifier}/available-contracts")]
-        public async Task<ActionResult<ApiCollection<ApiUnallocatedContract>>> GetProjectAvailableContracts([FromRoute]ProjectIdentifier projectIdentifier)
+        public async Task<ActionResult<ApiCollection<ApiUnallocatedContract>>> GetProjectAvailableContracts(
+            [FromRoute]ProjectIdentifier projectIdentifier,
+            [FromServices] IHttpClientFactory httpClientFactory,
+            [FromServices] IFusionContextResolver contextResolver)
         {
-            await Task.Delay(1);
+            var context = await contextResolver.ResolveContextAsync(ContextIdentifier.FromExternalId(projectIdentifier.ProjectId), FusionContextType.OrgChart);
 
-            var contracts = new[]
-            {
-                new ApiUnallocatedContract { ContractNumber = "0000000001" },
-                new ApiUnallocatedContract { ContractNumber = "0000000002" },
-                new ApiUnallocatedContract { ContractNumber = "0000000003" },
-                new ApiUnallocatedContract { ContractNumber = "0000000004" },
-                new ApiUnallocatedContract { ContractNumber = "0000055555" },
-                new ApiUnallocatedContract { ContractNumber = "0000666666" },
-                new ApiUnallocatedContract { ContractNumber = "1000000000" },
-                new ApiUnallocatedContract { ContractNumber = "1111111111" }
-            };
-
-            return Ok(new ApiCollection<ApiUnallocatedContract>(contracts));
+            if (context == null)
+                return ApiErrors.NotFound($"/contexts/project/{projectIdentifier.ProjectId}");
+
+            var projectMasterContext = await contextResolver.RelationsFirstOrDefaultAsync(context, FusionContextType.ProjectMaster);
+
+            if (projectMasterContext == null)
+                return ApiErrors.NotFound($"/contexts/projectMaster");
```

Comment:
Should we implement this as an extension method in the integration lib? 
Think getting from a org chart to project master would be a common operation.. Could also utilize the path query endpoint..

Could maybe for now create an extension class in the resources project, 

```
public static class ContextResolverExtensions 
{
    public static Task<FusionContext> ResolveProjectMasterAsync(this IFusionContextResolver resolver, ProjectIdentifier projectid) 
    { /* The implementation highligted */ }
}
```

Would improve readbility.

Reply:
Good suggestion, will do it.